sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -588,7 +588,9 @@ block_store_impl(T *p, simd<T, NElts> vals, simd_mask<1> pred, FlagsT flags) {
 
   constexpr int SmallIntFactor64Bit = sizeof(uint64_t) / sizeof(T);
   constexpr int SmallIntFactor32Bit =
-      std::max(static_cast<size_t>(1), sizeof(uint32_t) / sizeof(T));
+      sizeof(uint32_t) / sizeof(T) > static_cast<size_t>(1)
```

Comment:
nit: static_cast<size_t>(1) seems redundant here.
It was used in max() call to make the operand having same type and help compiler to find the matching max() prototype.